glean-core/ios/Glean/Utils.swift

```diff
@@ -11,13 +11,34 @@ extension String: Error {
     public var errorDescription: String? { return self }
 }
 
+/// Checks to see if a string matches a regex
```

Comment:
This doc string should be moved down to the `matches` function and the other newly introduced `var`s should be documented as well.